.github/workflows/ci.yml

```diff
@@ -58,12 +61,13 @@ jobs:
         security set-keychain-settings -lut 7200 temporary
     - name: Build-Test
       run: set -o pipefail && env NSUnbufferedIO=YES bundle exec rake test:macos
+      env:
+        DEVELOPER_DIR: ${{ env.CI_XCODE_13 }}
     - name: Send codecov
       run: bash <(curl https://codecov.io/bash)
 
   facebook_utils:
-    needs: macos
```

Comment:
The `needs: macos` was placed here to give 1) A higher priority to certain builds 2) timing, when these are organized the CI finishes all builds faster 3) When the higher priority builds fail, the lower priority don't run.

The above is extremely important since this repo has been moved to Github Actions (the Swift SDK was on Actions before this one) as there are limited resources for macOS dependent builds (5 systems across the whole organization). The Swift SDK along with the ParseLiveQuery-iOS repo share these resources with this SDK. So when builds fail, they should hopefully fail fast and give up the resources when they are not needed. This will needed when PR's are made across the organization around the same time.

If `macos` is considered a flakey build, I recommend switching this to `ios` or a higher priority build that's less flakey.

Reply:
That's fair, and I understand the motivation; the trouble is I'd estimate macOS and iOS builds have typically been about equally flakey. I'm not sure that the build time saved ends up actually saving time if we have to keep re-running the builds to see all the results.

Reply:
Conversely, if we attach these to a reliable test, they're still going to be running pretty much every time.

Reply:
If you end up removing flakey tests until they are fixed you can depend on macos or ios. You could also depend on `parseui` if that's less flakey or one of the other tests.

Reply:
> Conversely, if we attach these to a reliable test, they're still going to be running pretty much every time.

They will only run if the PR doesn't introduce a break to that build or test.

Reply:
> If you end up removing flakey tests until they are fixed you can depend on macos or ios.

It's a frustrating thing; there are some that seem to fail more often than others, but which ones tend to fail appears to change over time.

But some of them seem to just fail once, never to be reproduced. It's... not great.

Reply:
So yeah, I can make these depend on something that we'd expect to be stable like ParseUI. I suppose that could limit the amount of concurrent builds coming from this repo.

Reply:
> but I don't see it significantly reducing the total number of builds.

It matters the PR, maybe not in this current PR, but when others open up it can have a significant impact. It will also halt additional builds when someone submits bad code to `parseui`. In the future, when the flakey tests are resolved, the dependency in the CI should be moved back to `macos` or `ios` as I expect most code fixes in PR's will target the main source code and not parseui.

Reply:
For sure. I think if the tests weren't under the influence of some nebulous dark magic, I'd want to split them into 2 groups; Does It Work (iOS, macOS, ParseUI, TwitterUtils, FacebookUtils), and Does It Release (Release, Carthage, Cocoapods, Docs). 

I'd want the Does It Release group to depend on the Does It Work group. Release-type builds typically take the longest, and if the functionality is broken there's no need to see if the product can be released.